### DISCUSSION TITLE
Policy regarding typos in attribute names

### DISCUSSION THREAD

**Discussion by mdeweerd** [zigpy/zigpy#902] [ZHA development]
There are several typos in attribute names.

Changing them may have a functional impact for end-users and possibly also accross the library/quirks.

What's the policy:
- Keep the name errors until "end of life"
- Correct the name errors in the "ng" version
- Correct the name errors, but keep backward compatibility by maintaining the typos as alternatives.
- Correct tha name errors and ensure only updates to "zigpy" libraries and quirks.

If the policy allows it, I may provide PRs for correcting some of them.

**Comment by Hedda** [comment]
For reference, Home Assistant integration updates break due to fixes and changes all the time and just inform users via release notes, see ex:

https://www.home-assistant.io/blog/2022/02/02/release-20222/#breaking-changes

https://community.home-assistant.io/t/breaking-changes-and-their-impact-25-million-minutes-a-year/170700

https://community.home-assistant.io/t/what-is-up-with-all-these-breaking-changes/103134

**Reply by MattWestb** [reply]
Very often also HA have making changes and the automation is running but all entries is broken so then editing one working automation is empty and is braking if saving it without re doing all that is lost.

I think the best is making one PR and flagging it as braking and user must looking thru there automatons and correcting them if attribute names is being corrected.
Having ole bad names is not good in the long run and shall being corrected in one good maintained system.

**Reply by mdeweerd** [reply]
There is some effort in HA to warn ahead of time about breaking changes, and probaby not enough.

Personnally I think that not many users - if any - rely on the names of the attributes, and to me it's more hindering when you're looking for an attribue name (in the UI for instance) and you can't find by name because it's misspelled.

I was more thinking about having multiple entries in the table for the same attributes in an order that makes sure that the best attribute name is found by the code when converting and id to a name.   I suppose that this means that deprecated names would have to be added at the end of the list.

Further all deprecated names could be added to a new list "deprecated_attr_names" that could be checked in locations where attribute names are really used.  And warn (log file) that the attribute name is deprecated.

At the same time the "deprecated_attr_names" list could do the job of helping with backward compatibility by making it a dictionnary that would translate a deprecated name in the new name.  So when looking up an attribute, check the deprecated list to resolve to the correct name, and then lookup the attribute.

I do no expect deprecated name clashes accross the clusters as this concerns only misspelled attribute names.

So personnally I believe it's not needed to make it a breaking change.

However it's also a policy if one wants to support some backward compatibility, for how long, and how to label it in the code.

I am also not just making a PR without checking first because too many change proposals are refused for some kind of policy.   So I prefer checking up front.  Maybe making the changes in the current zigpy/zigpy is futile because of the upcoming 'ng' version.

**Comment by Adminiuga** [comment]
There's no policy per se. Depends on the attribute and how widespread it is being used. 
The primary effort is to make sure zha integration keeps working, and since it is using pinned versions, i don't see issues updating those
